### PR TITLE
Prepare shelf_web_socket 2.0 for release

### PR DESCRIPTION
**Pull request by brianquinlan**
- Remove "wip" from release label
- Accept web_socket_channel 2.x and 3.x

---

- [X] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/shelf_web_socket/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## 2.0.0-wip
+## 2.0.0
 
 * Require Dart `^3.0.0`.
 * **BREAKING:**: Remove support for hijacking WebSocket requests that are not
```
Comment (kevmoo) [inline pkgs/shelf_web_socket/CHANGELOG.md:4]: And this is good because we can delete a bunch of duplicate code, right?

Comment (brianquinlan) [reply pkgs/shelf_web_socket/CHANGELOG.md:4]: The code is already deleted: https://github.com/dart-lang/web_socket_channel/pull/341

Comment (kevmoo) [reply pkgs/shelf_web_socket/CHANGELOG.md:4]: but this work aligns with that work...hence why we're not as flexible?

Comment (brianquinlan) [reply pkgs/shelf_web_socket/CHANGELOG.md:4]: Yes. `package:web_socket_channel` has a partial copy of the `dart:io` WebSocket implementation that https://github.com/dart-lang/web_socket_channel/pull/341 removes. The code was the cause of the [recent security bug](https://github.com/dart-lang/web_socket_channel/issues/334).

**Review by natebosch — APPROVED** [review]

**Pull request merged**